resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/FileUtil.java

```diff
@@ -340,6 +347,71 @@ public static final void transferStreams(InputStream source, OutputStream destin
 		}
 	}
 
+	public static char[] readAllChars(IFile file) throws CoreException {
```

Comment:
Better place then in IFile as well, then it could be overwritten for local files as well.

Reply:
I don't understand what you mean since LocalFile is not in the type hierarchy of IFIle. I put that code in a Utils class so that the Interface is not polluted with code. Also i am not aware why it should be ever overridden.

Reply:
there is now `IFile.readAllBytes()` and in the same it should have `IFile.readAllChars()` that the might delegate to the (LocalFile)Store for a maybe better alternative ... this does not really "pollutes" the interface instead everyone requiring to reference a Util method does + it require us to maintain the Utilclass as API.

Alternativly you can follwo the Aproach of the JDK, there each Path has a "Filesystem" (here each IFIle has a FileStore) and the Util method effectivly delegate to the FileSystem, that way the Filesystem defines how things are performed and you don't need any readAlBytes() in the File API. Then of course it would be bst to place the Util next to IFile and maybe name it `IFiles` instead of FileUtils...)